Volume permission problem + New upsource version

tobwiens opened the pull request:
Upsource had no permission to write the data volumes because those were created by root. Only when creating a folder and then "applying" a volume onto that folder, the permission will be kept. 

Updated the README.md to not use the -v flag when creating the data container. But using the initial image.

tobwiens committed:
Create data, logs and backups directory and let it be owned by the upsource user. Then the upsource user will have access to the volumes. The VOLUME instruction is put after the creation of the folders, because otherwise the volumes will be owned by root.

tobwiens committed:
Use the klikatech/upsource image, because that will keep the upsource:upsource user permissions.

tobwiens committed:
When adding volumes after image creation, the user will switch to root again. Then the upsource user will not have access to it anymore. So as a solution don't specify the volumes here.

evolchek commented:
Thank you for pointing out the issue. However I'm not merging your fix as:
1. you have some mess in the Dockerfile (see line 14)
2. I would rather keep using the busybox image for data container as it is smaller. I've fixed the permissions problem. See readme.

tobwiens commented:
1. Okay I cleaned up the Dockerfile, thank you, I missed that.
2. I know it does not look that good when using the same container as a
   data container. But you should be aware that it takes more space with two
   different containers than with only one. The data-container will not change
   anything which means there is no additional disk-space or RAM acquired.
   Whereas when you take two different images, both are downloaded and it
   takes up more disk space.

Cheers

On 23 October 2015 at 15:54, Eugene Volchek notifications@github.com
wrote:

> Thank you for pointing out the issue. However I'm not merging your fix as:
> 1. you have some mess in the Dockerfile (see line 14)
> 2. I would rather keep using the busybox image for data container as it is
> smaller. I've fixed the permissions problem. See readme.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Klika-Tech/docker-upsource/pull/3#issuecomment-150579203
> .
